raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -211,6 +218,71 @@ contract MonitoringService is Utils {
         );
     }
 
+    function isAllowedToMonitor(
+        TokenNetwork token_network,
+        uint256 channel_identifier,
+        address closing_participant,
+        address non_closing_participant,
+        address monitoring_service_address
+    )
+        internal
+        returns (bool)
+    {
+        TokenNetwork.ChannelState channel_state;
+        uint256 settle_block_number;
+        (settle_block_number, channel_state) = token_network.getChannelInfo(
+            channel_identifier, closing_participant, non_closing_participant
+        );
+        require(channel_state == TokenNetwork.ChannelState.Closed);
+
+        // We don't actually know when the channel has been closed. So we'll
+        // make a guess so that assumed_close_block >= real_close_block.
+        uint256 assumed_settle_timeout = token_network.settlement_timeout_min();
+        require(settle_block_number >= assumed_settle_timeout);
+        uint256 assumed_close_block = settle_block_number - assumed_settle_timeout;
+        return block.number >= firstBlockAllowedToMonitor(
+            assumed_close_block,
+            assumed_settle_timeout,
+            closing_participant,
+            non_closing_participant,
+            msg.sender
+        );
+    }
+
+    function firstBlockAllowedToMonitor(
+        uint256 closed_at_block,
+        uint256 settle_timeout,
+        address participant1,
+        address participant2,
+        address monitoring_service_address
+    )
+        public pure
+        returns (uint256)
+    {
+        // avoid overflows when multiplying with percentages
+        require(settle_timeout < uint256(2**256 - 1) / 100);
+        require(closed_at_block < uint256(2**256 - 1) / 100);
```

Comment:
Just a thought & no need to react: I wondered how Solidity compiler understands `2**256`.  It's a complicated way to write zero, but it works and conveys the intuition.

Maybe I'll move `MAX_SAFE_UINT256` from `TokenNetwork.sol` into `Utils.sol`, and use the constant here.